android/src/main/java/com/iologinutils/AuthorizationManagerActivity.kt

```diff
@@ -0,0 +1,178 @@
+ * | Initiating +----->| Authorization +----->| Authorization  +----->| Redirect URI |
+ * |  Activity  |      |    Manager    |      |   Activity     |      |   Receiver   |
+ * |    (IO)    |<-----+   Activity    |<-----+ (e.g. browser) |      |   Activity   |
+ * |            | (C2) |               | (C1) |                |      |              |
+ * +------------+      +-+--+----------+      +----------------+      +-------+------+
+ *           ^            |     ^                                              |
+ *           |    (S3)    |     |                      (S2)                    |
+ *           +------------+     +----------------------------------------------+
+ *
+ *
+ * </pre>
+ *
+ * <p>The process begins with an activity requesting that an authorization flow be started,
+ * using {@link AuthorizationService#performAuthorizationRequest}.
+ *
+ * <ul>
+ *   <li>Step 1: Using an intent derived from {@link #createStartIntent}, this activity is
+ *       started. The state delivered in this intent is recorded for future use.
+ *   <li>Step 2: The authorization intent, typically a browser tab, is started. At this point,
+ *       depending on user action, we will either end up in a "completion" flow (S) or
+ *       "cancellation flow" (C).
+ *
+ *   <li>C flow:
+ *     <ul>
+ *       <li>Step C1: If the user presses the back button or otherwise causes the
+ *           authorization activity to finish, the AuthorizationManagementActivity will be
+ *           recreated or restarted.
+ *       <li>Step C2: The Promise contained in the AuthorizationService is rejected with the
+ *           with {@link IoLoginError#Type#NATIVE_AUTH_SESSION_CLOSED} error.
+ *     </ul>
+ *   <li>S flow:
+ *     <ul>
+ *       <li>Step S1: The authorization activity completes with a success of failure, and sends
+ *           this result to {@link RedirectUriReceiverActivity}.
+ *       <li>Step S2: {@link RedirectUriReceiverActivity} extracts the forwarded data, and
+ *           invokes AuthorizationManagementActivity using an intent derived from
+ *           {@link #createResponseHandlingIntent}. This intent has flag CLEAR_TOP set, which
+ *           will result in both the authorization activity and
+ *           {@link RedirectUriReceiverActivity} being destroyed, if necessary, such that
+ *           AuthorizationManagementActivity is once again at the top of the back stack.
+ *       <li>Step S3: The Promise contained in the AuthorizationService is resolved with the
+ *  *        received data.
+ *     </ul>
+ * </ul>
+ */
+class AuthorizationManagerActivity : Activity() {
+
+  private var authorizationStarted = false
+  private var authIntent: Intent? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    intent.extras?.let {
+      extractState(it)
+    } ?: savedInstanceState?.let {
+      extractState(it)
+    } ?: throw IllegalStateException("No state to extract")
```

Comment:
We need to find a better (more gentle) way to handle this situation. Maybe rejecting with a specific error. Better than crashing the whole app :). What do you think?

Reply:
You are right, we should reject with a platform specific error. We could extending the `AndroidError` with an error for this case.

Reply:
I've added an `IllegalStateException` rejection error in https://github.com/pagopa/io-react-native-login-utils/pull/11/commits/86956be4cd96830eda1f4d4d191137e182afb02a, let me know what do you think!

Reply:
Okay, perfect!